### ISSUE TITLE
App in dock doesn't work on ios13

### ISSUE THREAD

**Issue by Bilalmirza** [cocos2d/cocos2d-x#20118]
- cocos2d-x version: 3.17.2
- devices test on: ipad

 it doesn't work from dock anywhere.
Steps to Reproduce: 
1) Launch app.
2) double tap home. 
3) use dock to return to app.
4) does not respond

Video is attached.
https://www.dropbox.com/s/y1ep4m5sylo4yd0/IMG_1206.MOV?dl=0

**Comment by stevetranby** [comment]
Have you looked to see if there's any noticeably relevant Console messages.
(prob. not, but good first step since harder to debug while app not 'running')

**Comment by Bilalmirza** [comment]
good idea, This is what I'm getting

default	11:31:23.069709+0500	SpringBoard	Icon tapped: <private>
default	11:31:23.070605+0500	SpringBoard	Invalid property values passed in for event: iconTapped, values: <private>, registeredProperties: <private>
default	11:31:23.070719+0500	SpringBoard	Invalid property values passed in for event: iconTappedv2, values: <private>, registeredProperties: <private>

**Comment by stevetranby** [comment]
?? looks like either Apple bug or you need to adopt something for iOS 13. I know Screen instead of Application is a new thing (though more for document apps). Unfortunately, I've stayed away from iOS 13 and Catalina this round, so I'm waiting for stable release before bothering to upgrade (and to continue supporting iOS 8).

**Comment by stevetranby** [comment]
To rule out Apple bug you could create a new iOS app and a fresh new Cocos2d game and confirm the bug is Apple's or Cocos2d's, or that it's your specific game.

**Comment by Bilalmirza** [comment]
yep... thats the video of a fresh app attached in the first comment.
@stevetranby Apple actually approved my game with this bug. Its now available on apple arcade :D

**Comment by stevetranby** [comment]
Wait a sec, Apple Arcade :D
That's awesome, congrats!

**Comment by minggo** [comment]
@Bilalmirza it seems it is a bug of iOS? If so, then i will close this issue.

**Comment by Bilalmirza** [comment]
@minggo rest of the apps and games work fine in ios 13. Only cocos apps have this trouble

**Comment by minggo** [comment]
K, got it. We will take a look.

**Comment by Bilalmirza** [comment]
This is only occurring in some ipads... il'l make a list of ipads for you.

**Comment by anszom** [comment]
This may not be very relevant to you, but I've received a report of the same bug in a Qt-based iOS app. I've found your issue by googing the error messages:

> Oct  2 08:49:17 iPad SpringBoard(SpringBoardHome)[54] &lt;Notice&gt;: Icon tapped: &lt;private&gt;
> Oct  2 08:49:17 iPad SpringBoard(ProactiveEventTracker)[54] &lt;Notice&gt;: Invalid property values passed in for event: iconTapped, values: &lt;private&gt;, registeredProperties: &lt;private&gt;
> Oct  2 08:49:17 iPad SpringBoard(ProactiveEventTracker)[54] &lt;Notice&gt;: Invalid property values passed in for event: iconTappedv2, values: &lt;private&gt;, registeredProperties: &lt;private&gt;
> Oct  2 08:49:17 iPad SpringBoard(ProactiveEventTracker)[54] &lt;Notice&gt;: Invalid property values passed in for event: iconLaunched, values: &lt;private&gt;, registeredProperties: &lt;private&gt;
> Oct  2 08:49:17 iPad SpringBoard(ProactiveEventTracker)[54] &lt;Notice&gt;: Invalid property values passed in for event: iconLaunchedv2, values: &lt;private&gt;, registeredProperties: &lt;private&gt;
> Oct  2 08:49:17 iPad trustd[105] &lt;Notice&gt;: cert[0]: SubjectCommonName =(leaf)[]&gt; 0
> Oct  2 08:49:17 iPad trustd[105] &lt;Notice&gt;: cert[0]: LeafMarkerOid =(leaf)[force]&gt; 0
> Oct  2 08:49:17 iPad trustd[105] &lt;Notice&gt;: cert[0]: IssuerCommonName =(path)[]&gt; 0
> Oct  2 08:49:17 iPad SpringBoard(Security)[54] &lt;Notice&gt;: Trust evaluate failure: [leaf IssuerCommonName LeafMarkerOid SubjectCommonName]
> Oct  3 08:49:17 iPad SpringBoard(SpringBoardHome)[54] &lt;Notice&gt;: Icon touch canceled (tap gesture may still succeed): &lt;private&gt;

Unfortunately I don't have access to the problematic device.
